src/main/java/uk/gov/ons/ctp/response/sample/ingest/CsvIngesterBusiness.java

```diff
@@ -85,44 +105,68 @@ public CsvIngesterBusiness() {
     columnPositionMappingStrategy.setColumnMapping(COLUMNS);
   }
 
+  private String generateErrorReport(int maxErrors, List<BusinessSampleUnit> sampleUnitList, List<ValidationError> errors){
+    StringBuffer report = new StringBuffer();
+
+    report.append(String.format("timestamp: %s, valid samples: %d, errors: %d (fail at %d errors)\n",
+            new Date().toString(), sampleUnitList.size(), errors.size(), maxErrors));
+    report.append(errors
+        .stream()
+        .map(ve -> String.format("Line %d: %s -> %s", ve.getLineNumber(), ve.getErrorType().name(), ve.getErrorDetail()))
+        .collect(Collectors.joining("\n")));
+
+    return report.toString();
+  }
+
+  private void validateFilename(String filename) throws IngesterCTPException {
```

Comment:
Maybe this could live in the end point and validate all sample types. They're all CSV after all?

Reply:
Done